### PR TITLE
Fix pause controller

### LINKED ISSUE #476: Fix pause
As a player, I want to be able to pause the game, so that I can take a quick break when necessary.

Currently, when the game is paused and resumed, the framerate drops by about 30%, and memory usage doubles, and other random problems occur. This task is to fix the pause, so that it no longer impacts game performance.

### PR DESCRIPTION
**Pull request by Taeir**
- Removed pause controller
- Added new pausing logic that simply suppresses updates after 2 ticks.
- Fixed tests

This closes #476.

### CODE REVIEW

File: src/main/java/nl/tudelft/contextproject/controller/GameController.java [hunk 1]
```diff
@@ -49,7 +50,11 @@ public void initialize(AppStateManager stateManager, Application app) {
 
 	@Override
 	public GameState getGameState() {
-		return GameState.RUNNING;
+		if (Main.getInstance().isPaused()) {
+			return GameState.PAUSED;
+		} else {
+			return GameState.RUNNING;
+		}
```
Comment (ChielBruin) [inline src/main/java/nl/tudelft/contextproject/controller/GameController.java]: Shouldn't this if be in the main class?

**Comment by ChielBruin** [comment]
I made one little comment. The rest looks good!

File: src/main/java/nl/tudelft/contextproject/controller/GameController.java (hunk 1 above)
Comment (Taeir) [inline src/main/java/nl/tudelft/contextproject/controller/GameController.java]: Good point I will fix it.

File: src/main/java/nl/tudelft/contextproject/Main.java [hunk 2]
```diff
@@ -327,6 +322,26 @@ public void moveCameraTo(Vector3f newLoc) {
...
+			super.update();
+		} else if (paused == 1 || paused == 2) {
+			super.update();
+			paused++;
+		} else {
+			runQueuedTasks();
+			getInputManager().update(1f);
+			
+	        try {
+	            Thread.sleep(50); // throttle the CPU when paused
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/Main.java:336]: Good comment, I was wondering why we would need a sleep.

**Comment by Geweldig** [comment]
Nice and clean addition to the code, merging this.

**Pull request merged**